src/ranch.erl

```diff
@@ -272,6 +274,63 @@ procs1(Ref, Sup) ->
 		[]
 	end.
 
+-spec wait_for_connections(ref(), '>' | '>=' | '==' | '=<', non_neg_integer()) -> ok;
+	(ref(), '<', pos_integer()) -> ok.
+wait_for_connections(Ref, Op, NumConns) ->
+	wait_for_connections(Ref, Op, NumConns, 1000).
+
+-spec wait_for_connections(ref(), '>' | '>=' | '==' | '=<', non_neg_integer(),
+	non_neg_integer()) -> ok;
+	(ref(), '<', pos_integer(), non_neg_integer()) -> ok.
+wait_for_connections(Ref, Op, NumConns, Interval) ->
+	if
+		is_integer(Interval) andalso Interval>=0 ->
+			ok;
+		true ->
+			error(badarg)
+	end,
+	if
+		is_integer(NumConns) andalso NumConns>=0 ->
+			ok;
+		true ->
+			error(badarg)
+	end,
+	case Op of
+		'>' ->
+			ok;
+		'>=' ->
+			ok;
+		'==' ->
+			ok;
+		'=<' ->
+			ok;
+		'<' when NumConns>=1 ->
+			ok;
+		_ ->
+			error(badarg)
+	end,
```

Comment:
Not quite as expected lol. I'll merge and make it more to what I was thinking.

Reply:
I know, I'm not exceedingly proud of this. Have been struggling to find a good solution style-wise, but mostly failed for one reason or another. Finally settled for this one. Feel free to change it to your liking :)